src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -607,6 +608,8 @@ export class NotebookModel extends Disposable implements INotebookModel {
 						partialSource = source.slice(selection.startLineNumber - 1, selection.startLineNumber)[0].slice(0, selection.startColumn - 1);
 						headsource = headsource.concat(partialSource.toString());
 					}
+					// save attachments before updating the cell contents
+					attachments = this.cells[index].attachments;
```

Comment:
And are you removing the attachments that no longer exist in the original cell?

Reply:
There's not need to remove them because we are updating the original cell source, and it removes the unused attachments here. https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/services/notebook/browser/models/cell.ts#L337

Reply:
Oh cool, a couple things : 

1. Please say that in a comment here
2. Look into re-using the logic for that cleanup function in the new "get cell attachments" function you have. At the very least this regex there should be used in both places to find the attachment keys in the source and can be shared

Reply:
@barbaravaldez Make sure you add the comment about how we're not modifying the original cell because it already goes through the cleanup process when the source is changed.